js/apps/events-page/controllers/eventsPageCtrl.js

```diff
@@ -1,39 +1,212 @@
 ((app) => {
 	'use strict';
 
-	const EventsPageCtrl = function EventsPageCtrl($scope, $timeout, CONSTANTS, eventsService, dateUtility) {
+	const EventsPageCtrl = function EventsPageCtrl($scope, $timeout, $animate, CONSTANTS, eventsService) {
 		const self = this;
 		const firstPage = 1;
+		const startDateLocaleString = moment().format();
+		const endDate = moment().add(30, 'd');
+		const endDateLocaleString = endDate.format();
 		const requestModel = {
-			StartDate: new Date(),
-			EndDate: new Date(),
+			StartDate: startDateLocaleString,
+			EndDate: endDateLocaleString,
 			Page: firstPage,
-			IsOngoingVisible: true
+			IsOngoingVisible: true,
+			IsSpacesReservationVisible: false,
+			Limit: CONSTANTS.requestChunkSize,
+			EventsTypes: [],
+			AgeGroups: [],
+			Locations: []
+		};
+		const eventDateBarStickySettings = {
+			zIndex: 100,
+			responsiveWidth: true
 		};
 
 		/* ** Public ** */
 
 		self.eventGroups = [];
+		self.keywords = '';
+		self.chunkSize = CONSTANTS.requestChunkSize;
+		self.totalResults = 0;
+		self.isLastPage = false;
+		self.areDatesInvalid = false;
+		self.isLoading = true;
+		self.hasResults = true;
+
+		self.locations = requestModel.Locations;
+		self.eventsTypes = requestModel.EventsTypes;
+		self.ageGroups = requestModel.AgeGroups;
+		
+		self.keywordSearch = () => {
+			requestModel.Keyword = self.keywords;
+			requestModel.StartDate = startDateLocaleString;
+			requestModel.Page = 1;
+			self.eventGroups = [];
+			self.hasResults = true;
+			self.isLoading = true;
+
+			eventsService.get(requestModel).then(processEvents);
+		};
+
+		self.filterByDate = () => {
+			if (isDateRangeValid(self.userStartDate, self.userEndDate)) {
```

Comment:
set the conditional to self.areDatesInvalid above the if.

```self.areDatesInvalid = !isDateRangeValid(...)```

Reply:
done!